src/lib/plugins/system-info.js

```diff
@@ -0,0 +1,517 @@
+// This library contains several chunks of code from Sebastian Hildebrandt's
+// systeminformation library, MIT
+// https://github.com/sebhildebrandt/systeminformation
+
+/* global process */
+
+import os from 'os'
+import fs from 'fs'
+import { exec } from 'child_process'
+
+const PLATFORM = os.type(),
+  LINUX = (PLATFORM == 'Linux'),
+  DARWIN = (PLATFORM == 'Darwin'),
+  WINDOWS = (PLATFORM == 'Windows_NT'),
+  NOT_SUPPORTED = 'not supported',
+  MATCH_SPACES = / +/g,
+  NETWORK = {}
+
+let defaultInterface
+
+const isFunction = fn =>
+  fn && {}.toString.call(fn) === '[object Function]'
+
+const splitOutput = output => output.toString().split('\n')
+
+const splitWithAndTrim = (output, pattern) => output.split(pattern)[1].trim()
+
+export const time = () => ({
+  current: Date.now(),
+  uptime: os.uptime()
+})
+
+/**
+ * Gather the memory systema information
+ *
+ * @param {function} callback Fallback function
+ * @returns Promise
+ */
+export const mem = callback =>
```

Comment:
If we do decide to keep this code, I think we should stick to Promises everywhere in our codebase. It would be good to remove the callback argument from this and other functions in this file :)